pkg/syncer/syncer.go

```diff
@@ -299,6 +324,35 @@ func (c *Controller) handleErr(err error, i interface{}) {
 	klog.Errorf("Dropping key %q after failed retries: %v", i, err)
 }
 
+const nsDelimiter = "--"
+const nsTemplate = "kcp" + nsDelimiter + "%s" + nsDelimiter + "%s"
+
+func transformForCluster(gvr schema.GroupVersionResource, meta metav1.Object) (string, string, error) {
+	if gvr.Group == "" && gvr.Resource == "namespaces" {
+		return fmt.Sprintf(nsTemplate, meta.GetClusterName(), meta.GetName()), "", nil
+	}
+	return meta.GetName(), fmt.Sprintf(nsTemplate, meta.GetClusterName(), meta.GetNamespace()), nil
```

Comment:
Namespaces have a length limit, we likely want to do something else here. Perhaps hash the inputs and prefix the output, if that gives us enough collision resistance?

Reply:
We need to be able to reverse the mapping so we can sync the status up to kcp.  So we can either do that by having a reversible transformation (like this PR) or by storing the mapping somewhere (an annotation?).

I was thinking we'd do the simplest thing for now and follow up with a better transformation strategy, but I am happy to do the better transformation in this PR.